p2p/net/swarm/swarm_listen.go

```diff
@@ -13,58 +14,56 @@ import (
 	manet "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net"
 	ps "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-peerstream"
 	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
-	multierr "github.com/ipfs/go-ipfs/thirdparty/multierr"
 )
 
-// Open listeners for each network the swarm should listen on
-func (s *Swarm) listen(addrs []ma.Multiaddr) error {
-
+// Open listeners and reuse-dialers for the given addresses
+func (s *Swarm) setupAddresses(addrs []ma.Multiaddr) error {
 	for _, addr := range addrs {
 		if !addrutil.AddrUsable(addr, true) {
 			return fmt.Errorf("cannot use addr: %s", addr)
 		}
 	}
 
-	retErr := multierr.New()
+	dialer := manet.Dialer{Dialer: net.Dialer{Timeout: DialTimeout}}
+	for _, a := range addrs {
+		switch {
+		case conn.IsTcpMultiaddr(a):
+			tpt, err := conn.NewTcpReuseTransport(dialer, a)
+			if err != nil {
+				return err
```

Comment:
if listening to any of the listener addresses fails today, ipfs fails to start. This behaviour is EXACTLY the same.

this returns an error if any of the listens fail: https://github.com/ipfs/go-ipfs/blob/d4938c9b0cd57c52c71ec83998a8288c3247826c/p2p/net/swarm/swarm_listen.go#L43

That error gets propogated all the way back up and breaks init.

The 'good and bad' addresses test only passes because the bad addresses get filtered out here: https://github.com/ipfs/go-ipfs/blob/d4938c9b0cd57c52c71ec83998a8288c3247826c/p2p/net/swarm/swarm.go#L76